controller/GiPlanned.controller.js

```diff
@@ -0,0 +1,408 @@
+sap.ui.define([
+	"sap/ui/core/mvc/Controller",
+	"sap/m/MessageToast",
+	"sap/ui/model/json/JSONModel",
+	"sap/m/MessageBox"
+], function(Controller, MessageToast, JSONModel, MessageBox) {
+	"use strict";
+
+	var GlobalFuncs = sap.ui.controller("smud.org.wmLM01.controller.GlobalFuncs");
+	var zController;
+
+	return Controller.extend("smud.org.wmLM01.controller.GiPlanned", {
+
+		/**
+		 * Called when a controller is instantiated and its View controls (if available) are already created.
+		 * Can be used to modify the View before it is displayed, to bind event handlers and do other one-time initialization.
+		 * @memberOf smud.org.wmLM01.view.GiPlanned
+		 */
+		onInit: function() {
+			zController = this;
+			this.getView().setModel(sap.ui.getCore().getModel("GiPlanned"));
+			this.createTable();
+
+			this._oView = this.getView();
+			this._oView.addEventDelegate({
+				onBeforeHide: function(oEvent) {
+					// Remove items from the table and clear filter search on BACK
+					sap.ui.getCore().byId(zController.createId("materialTable")).removeAllItems();

```

Comment:
why we are always using the core? can't we just do a simple `this.byId("materialTable")`?